python/az/aro/azext_aro/custom.py

```diff
@@ -134,9 +137,43 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
                        parameters=oc)
 
 
-def aro_delete(client, resource_group_name, resource_name, no_wait=False):
+def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
     # TODO: clean up rbac
 
+    try:
+        cluster_info = client.get(resource_group_name, resource_name)
+    except CloudError:
+        pass
+
+    if cluster_info:
+        master_subnet = cluster_info.master_profile.subnet_id
+        worker_subnets = [w.subnet_id for w in cluster_info.worker_profiles]
+
+        master_parts = parse_resource_id(master_subnet)
+        vnet = resource_id(
+            subscription=master_parts['subscription'],
+            resource_group=master_parts['resource_group'],
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=master_parts['name'],
+        )
+
+        aad = AADManager(cmd.cli_ctx)
+
+        client_id = cluster_info.service_principal_profile.client_id
+        client_sp = aad.get_service_principal(client_id)
```

Comment:
get_service_principal returning None is a feasible failure mode both here and below, which cannot be recovered by this code, and it might make sense to catch it and print a clearer error